### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by BouaikIlham**
## In this pull request :
  - [ ] Add the  validations: 
       - For the User model:
            - Name must not be blank.
            - PostsCounter must be an integer greater than or equal to zero.
       - For the Post model:
            - Title must not be blank.
            - Title must not exceed 250 characters.
            - CommentsCounter must be an integer greater than or equal to zero.
            - LikesCounter must be an integer greater than or equal to zero.
- [ ] Add unit specs for  models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    subject.likescounter = -2
+    expect(subject).to_not be_valid
+  end
+  it 'title must not exceed 250 characters' do
+    subject.title = 'orem ipsum dolor sit amet, consectetuer adipiscing elit.
+    Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes,
+    nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium q'
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/post_spec.rb]: - [ ] Kindly add a test for update_counter and recent_comments methods. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.)

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  it 'postCounter must be integer greater than or equal zero' do
+    subject.postsCounter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'postCounter must be integer greater than or equal zero' do
+    subject.postsCounter = 0
+    expect(subject).to be_valid
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/user_spec.rb]: - [ ] Kindly add a test for return_recent_posts method. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @BouaikIlham  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- You added validations to models according to requirements. 
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] You didn't create a unit test for like and comment models.The like and comment model has custom methods. Kindly create it.

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by DammyShittu — APPROVED** [review]
Hi @BouaikIlham,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**